cmd/kube-apiserver/app/server.go

```diff
@@ -92,25 +103,102 @@ cluster's shared state through which all other components interact.`,
 
 // Run runs the specified APIServer.  This should never exit.
 func Run(s *options.ServerRunOptions) error {
+	install.Install(api.GroupFactoryRegistry, api.Registry, api.Scheme)
 	config, sharedInformers, err := BuildMasterConfig(s)
 	if err != nil {
 		return err
 	}
 
-	return RunServer(config, sharedInformers, wait.NeverStop)
+	return RunServer(s, config, sharedInformers, wait.NeverStop)
 }
 
 // RunServer uses the provided config and shared informers to run the apiserver.  It does not return.
-func RunServer(config *master.Config, sharedInformers informers.SharedInformerFactory, stopCh <-chan struct{}) error {
+func RunServer(s *options.ServerRunOptions, config *master.Config, sharedInformers informers.SharedInformerFactory, stopCh <-chan struct{}) error {
+	// if you're running the aggregator, you have to shut off root discovery
+	config.GenericConfig.EnableDiscovery = false
+
 	m, err := config.Complete().New()
 	if err != nil {
 		return err
 	}
 
+	// create the aggregator config based on the kubeapiserver config
+	// insane doesn't not even begin to describe this.  Turns out that you need protobufs to run with the
+	// storagefactory for kube.  this wires us to a different one
+	s.Etcd.StorageConfig.Codec = aggregatorapiserver.Codecs.LegacyCodec(schema.GroupVersion{Group: "apiregistration.k8s.io", Version: "v1alpha1"})
+	s.Etcd.StorageConfig.Copier = aggregatorapiserver.Scheme
+	config.GenericConfig.RESTOptionsGetter = &genericoptions.SimpleRestOptionsFactory{*s.Etcd}
+	client, err := kubeclientset.NewForConfig(config.GenericConfig.LoopbackClientConfig)
+	if err != nil {
+		return err
+	}
+	// if the files are there, load them. This makes things nice and easy for my demo
+	certBytes, _ := ioutil.ReadFile("/var/run/kubernetes/client-auth-proxy.crt")
```

Comment:
@lavalamp These will eventually be flags that are passed through.  It is a client cert which is signed by the requestheader-ca key.  My experience with RBAC was that the hardest part was actually plumbing those options through.

Want to get started now?  You could work on getting the requestheader CA and a client cert/key pair signed and local to the kube-apiserver.